numpy/distutils/command/build_ext.py

```diff
@@ -570,7 +570,7 @@ def _process_unlinkable_fobjects(self, objects, libraries,
         unlinkable_fobjects = list(unlinkable_fobjects)
 
         # Expand possible fake static libraries to objects
-        for lib in libraries:
+        for lib in libraries[:]:
```

Comment:
Can you add a comment above this line that `[:]` is on purpose, to not modify the original list? Otherwise the next over-eager code cleanup may undo the fix again.

Reply:
Added an explanation is latest push, feel free to resolve or suggest different wording